### PR TITLE
Fixes for shutting down pacemaker resources

### PR DESCRIPTION
**Pull request by jsuchome**
Not yet tested, still investigating it this can help...

### CODE REVIEW

File: chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb [hunk 1]
```diff
@@ -104,7 +104,7 @@
       for type in clone ms primitive; do
         for resource in $(crm configure show | grep ^$type | cut -d " " -f2);
         do
-          crm resource stop $resource
+          crm --force --wait resource stop $resource
```
Comment (aplanas) [inline chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb]: Is not `--wait` alone enough? This `--force` is for user interaction, is happening with this `--wait` option?

Comment (jsuchome) [inline chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb]: It's true that `--force` was likely not needed so far. But it's probably better to be sure if some user interaction would be required...

**Comment by aplanas** [comment]
+1

File: chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb (hunk 1 above)
Comment (vuntz) [inline chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb]: We actually always need `--force`, it was a bug to not have it. The `--wait` is the imrpovement here

**Comment by vuntz** [comment]
+1

**Comment by jsuchome** [comment]
New significant commit -> resetting your +1 counts

File: chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb [hunk 2]
```diff
@@ -31,6 +31,18 @@
...
+  bash "stop rabbitmq resources" do
+    code <<-EOF
+        for resource in g-rabbitmq rabbitmq fs-rabbitmq ms-drbd-rabbitmq drbd-rabbitmq;
+        do
+          crm --force --wait resource stop $resource
+        done
+      done
+    EOF
+    only_if { ::File.exist?("/usr/sbin/crm") }
+  end
```
Comment (vuntz) [inline chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb]: Oh, I thought you would do it in the other recipe. But okay :-)

Comment (sayalilunkad) [inline chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb]: I am not completely sure but I think it is possible to use execute blocks here instead similar to how it is done here https://github.com/crowbar/crowbar-core/pull/301

**Comment by jsuchome** [comment]
So, first testing shows that the resources were stopped and deleted without any problem. I'll try yet different setup...

**Comment by vuntz** [comment]
+1 for this, although I believe it's fine to put the new bash resource in stop-services-before-upgrade.rb instead (and avoid code duplication).

**Comment by vuntz** [comment]
Still +1.

**Comment by aplanas** [comment]
+1

**Comment by vuntz** [comment]
@jsuchome are you okay merging? (You might want to force push to fix the hound comment and just merge without waiting again for gating).

**Comment by jsuchome** [comment]
@vuntz rebased and fixed hound issue; but I can't merge it myself before another gating passes

**Pull request merged**